matrix/main.py

```diff
@@ -132,6 +132,13 @@ def setup(matrix, args=None):
     parser.add_argument("-k", "--keep-models", action="store_true",
                         default=False, help="Keep the created per-test models "
                                             "for further inspection")
+    parser.add_argument("-r", "--deployer", default="python-libjuju",
+                        help="Method to use to deploy charms. Defaults to "
+                        "'python-libjuju'. Other options include 'congure-up'."
+                        " Note that if you choose conjure-up, --path "
+                        "should point to a conjure-up spell "
+                        "(with metadata.yaml) "
+                        "instead of to a vanilla juju bundle.")
```

Comment:
@mikemccracken @battlemidget I added logic so that we detect whether or not something is a spell based on it having a metadata.yaml with a friendly-name attribute (version doesn't seem to be required). Thx for the clarification on what is and isn't a spell.